scripts/build/file2hex.py

```diff
@@ -65,6 +65,26 @@ def make_string_literal(chunk):
     print(''.join("\\x" + str(x) for x in hexlist), end='')
 
 
+def process_content(content, length):
+    if args.format == "literal":
+        print('"', end='')
+        process_chunk_func = make_string_literal
+    else:
+        process_chunk_func = make_hex
+
+    if length < 0:
+        for chunk in iter(lambda: content.read(1024), b''):
+            process_chunk_func(chunk)
```

Comment:
I acknowledge that in this case, the use of dynamic calls reduces the number of branches.
And setting of `process_chunk_func` is close to its use, making it easy to follow.

However, code always evolves, so there is a risk that at some point setting `process_chunk_func` and the code using it can be located further apart, and thus make code harder to read and understand.

Also, if one of the functions in future needs an extra argument then both functions must takes such arg, even if unused.

Although the above obsdervations, then this approach is acceptable to me, but before given a +1, I would like to hear the opinion from other reviewers, as they might have different opinions.